resources/gcloud.rb

```diff
@@ -9,90 +9,286 @@
 end
 
 action :install do
-  platform_cmd = Mixlib::ShellOut.new('uname')
-  platform_cmd.run_command
-  platform_cmd.error!
-  platform = platform_cmd.stdout.strip.downcase
-
-  arch_cmd = Mixlib::ShellOut.new('uname -m')
-  arch_cmd.run_command
-  arch_cmd.error!
-  arch = arch_cmd.stdout.strip
-
-  version = new_resource.version
-  if version.empty?
+  # Check infomation of operation system
+  platform = platform()
+  arch = arch()
+  arch_core = arch_core(arch)
+
+  # version will be installed
+  latest_version = latest_version()
```

Comment:
I think it's not completely clean, I'm pretty messed up, and it's not good. I'm editing a few things to ensure we get a good release version before 16/8.